src/Symfony/Component/HttpClient/EventSource/MessageEvent.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Soyuka\EventSource;
+
+class MessageEvent
+{
+    public $data;
+    public $lastEventId;
+    public $eventName;
+    public $origin;
+
+    public function __construct($data = '', $lastEventId = null, $eventName = '', $origin = null)
+    {
+        $this->data = $data;
+        $this->lastEventId = $lastEventId;
+        $this->eventName = $eventName;
+        $this->origin = $origin;
+    }
+
+    public static function parse($data): ?self
+    {
+        $list = preg_split('/\r\n|\r|\n/', $data);
```

Comment:
Could be simplified:

```suggestion
        $list = preg_split('/[\r\n]/', $data);
```

Reply:
That's not the same: the current code handle `\r\n` as a single token, the suggestion splits between `\r` and `\n`.